osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySpinner.cs

```diff
@@ -45,36 +47,67 @@ private void load(DrawableHitObject drawableHitObject, ISkinSource source)
 
             DrawableSpinner = (DrawableSpinner)drawableHitObject;
 
-            AddRangeInternal(new[]
+            Container overlayContainer;
+
+            AddInternal(overlayContainer = new Container
             {
-                spin = new Sprite
+                Depth = float.MinValue,
+                RelativeSizeAxes = Axes.Both,
+                Children = new Drawable[]
                 {
-                    Anchor = Anchor.TopCentre,
-                    Origin = Anchor.Centre,
-                    Depth = float.MinValue,
-                    Texture = source.GetTexture("spinner-spin"),
-                    Scale = new Vector2(SPRITE_SCALE),
-                    Y = SPINNER_TOP_OFFSET + 335,
-                },
-                clear = new Sprite
-                {
-                    Alpha = 0,
-                    Anchor = Anchor.TopCentre,
-                    Origin = Anchor.Centre,
-                    Depth = float.MinValue,
-                    Texture = source.GetTexture("spinner-clear"),
-                    Scale = new Vector2(SPRITE_SCALE),
-                    Y = SPINNER_TOP_OFFSET + 115,
-                },
+                    spin = new Sprite
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.Centre,
+                        Texture = source.GetTexture("spinner-spin"),
+                        Scale = new Vector2(SPRITE_SCALE),
+                        Y = SPINNER_TOP_OFFSET + 335,
+                    },
+                    clear = new Sprite
+                    {
+                        Alpha = 0,
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.Centre,
+                        Texture = source.GetTexture("spinner-clear"),
+                        Scale = new Vector2(SPRITE_SCALE),
+                        Y = SPINNER_TOP_OFFSET + 115,
+                    },
+                }
             });
+
+            bonusCounter = (LegacySpriteText)source.GetDrawableComponent(new HUDSkinComponent(HUDSkinComponents.ScoreText));
+
+            if (bonusCounter != null)
```

Comment:
Initially, I was planning to make a PR to be able to construct `LegacySpriteText`s on each usage pretty simply, without having to call `GetDrawableComponent(ScoreText)` and guard against null.

But now I realized the legacy piece of hitcircle number fall backs to an `OsuSpriteText` instead.

Therefore I've posted this comment to ask: is it intentional making legacy skins fall back to using `OsuSpriteText` when no font exists?

I've [blamed through](https://github.com/ppy/osu/pull/3500/commits/0d8276c5f86a811b49d252271d6781f5ab48b708) but found nothing answering my question.

Reply:
explain. what does the legacy hitcircle have to do with this? can `bonusCounter` ever actually be null, and if yes, when? if it is possible for it to be null, then i'd say *something* has to be used, instead of just not showing the text.

Reply:
> what does the legacy hitcircle have to do with this?

they both have legacy sprite texts using legacy fonts which may or may not exist, I'm questioning where it's intentional making legacy sprite texts fallback to default osu!lazer text if no font exists.

> can `bonusCounter` ever actually be null, and if yes, when?

yes it can be null, when the font textures for the skin aren't provided, determined via `this.HasFont(font)`.

> if it is possible for it to be null, then i'd say something has to be used, instead of just not showing the text.

and I was asking whether it should not show anything for simplicity until falling back to default legacy skin is supported, or show a default osu!lazer text instead

but now that you said that, I guess it's more simple to do the latter, just need to expose a `FixedWidth` from within skinnable sprite text. will do on same PR.

Reply:
it it turns out to be null, this is going to throw, no? you're doing a cast on the previous line.

Reply:
you mean the cast to `LegacySpriteText`?

Reply:
It looks to not crash, but my brain can't comprehend this so I'm going to refactor it a touch.

Reply:
Fair, maybe worth a warning if possible then.